### PR TITLE
Template matching bug

### PR DESCRIPTION
**Pull request by h-mayorquin**
As discussed today, I did look into this. After looking into it, it seems that the problem is in the waveform extractor. If you look at the new assertion that I put at the beginning of the test, all the methods are failing (not just circus**). And the reason seems to be that the waveform_extractor is not returning any waveforms.** 

Why could this be? I think it is related to the changes that I did in mearec but I know less about the waveforme extractors as it is an object that I have never look into detail.

Maybe at this point something rings a bell, @alejoe91?

Otherwise, I will dig deeper.

### CODE REVIEW

File: src/spikeinterface/sortingcomponents/tests/test_template_matching.py [hunk 1]
```diff
@@ -13,55 +13,62 @@
...
+
+@pytest.mark.parametrize("method", matching_methods.keys())
+def test_find_spikes_from_templates(method, waveform_extractor):
+
+
+
+    recording = waveform_extractor._recording
+    waveform_extractor.get_waveforms("#0")
+    num_waveforms, _, _ = waveform_extractor.shape
+    assert num_waveforms != 0
```
[the remark this answers is not in the document]
Comment (alejoe91) [reply src/spikeinterface/sortingcomponents/tests/test_template_matching.py:42]: Thanks Ramon, nothing rings a bell..but at least we know where the error is coming from. @samuelgarcia do you have time to reproduce and fix it tomorrow?

**Comment by alejoe91** [comment]
It needs to be caused either by #1543 or #1599 (these are the PRs that changed the waveform extractor and we're merged in the last couple of days)

**Review by alejoe91 — APPROVED** [review]

File: src/spikeinterface/extractors/neoextractors/neobaseextractor.py [hunk 2]
```diff
@@ -264,22 +259,30 @@ def __init__(self, block_index=None,
             # use interger based unit_ids
             unit_ids = np.arange(spike_channels.size, dtype='int64')
 
+        guess_t_start_from_signal = False
+        stream_index = None 
+        if sampling_frequency is None:
+            sampling_frequency, stream_id = self._auto_guess_sampling_frequency()
+            stream_index = np.where(self.neo_reader.header["signal_streams"]["id"] == stream_id)[0][0]
+            guess_t_start_from_signal = True
+
```
[the remark this answers is not in the document]
Comment (h-mayorquin) [reply src/spikeinterface/extractors/neoextractors/neobaseextractor.py:268]: Why not?  It is not going to have any effect on how the frames are calculated but will make that information available if it exists.

[the remark this answers is not in the document]
Comment (samuelgarcia) [reply src/spikeinterface/extractors/neoextractors/neobaseextractor.py:268]: self._auto_guess_sampling_frequency() is a bad stuff that should be avoid when not necessary.

[the remark this answers is not in the document]
Comment (h-mayorquin) [reply src/spikeinterface/extractors/neoextractors/neobaseextractor.py:268]: That orthogonal to the question. The question is, should t start be extracted if possible for both cases: when the units are returned in frames and when not. Right now it is only extract for the case when no sampling frequency is provided. You want to reduce it even further. We can avoid the auto guess but still extract the t-start when available.

[the remark this answers is not in the document]
Comment (alejoe91) [reply src/spikeinterface/extractors/neoextractors/neobaseextractor.py:268]: @samuelgarcia even if  `handle_spike_frame_directly=True`, `sampling_frequency` is needed. So if it's None, we need to auto-guess it!

[the remark this answers is not in the document]
Comment (alejoe91) [reply src/spikeinterface/extractors/neoextractors/neobaseextractor.py:268]: I don't mind. If `handle_spike_frame_directly=True`, `self._t_start` will never be used, but it'll just be avilable as a hidden variable..fine for me!

[the remark this answers is not in the document]
Comment (h-mayorquin) [reply src/spikeinterface/extractors/neoextractors/neobaseextractor.py]: OK, sam, I did the changes you requested. My question still is, is this the result you want:


 get_t_start_from_signal?  | neo_returns_timestamps | neo_return_frames
-- | -- | --
sampling_frequency_provided? |  -- | --  
True | No | No
False | Yes | No

Thi is the logic implemented right now.

**Comment by alejoe91** [comment]
@samuelgarcia didnyou check this?

**Comment by samuelgarcia** [comment]
I am on it

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**